system-reinstall-bootc/src/lvm.rs

```diff
@@ -0,0 +1,103 @@
+use std::process::{Command, Stdio};
+
+use anyhow::Result;
+use bootc_mount::run_findmnt;
+use bootc_utils::CommandRunExt;
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct Lvs {
+    report: Vec<LvsReport>,
+}
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct LvsReport {
+    lv: Vec<LogicalVolume>,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub(crate) struct LogicalVolume {
+    lv_name: String,
+    lv_size: String,
+    lv_path: String,
+    vg_name: String,
+}
+
+pub(crate) fn binary_available(binary: &str) -> bool {
+    Command::new("which")
+        .arg(binary)
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .status()
+        .map(|status| status.success())
```

Comment:
Hmm it's a bit unusual to *run* the binary to find out if it's present. Hmmmm...this is pretty trivial functionality BUT there is a crate https://crates.io/crates/which that skimming seems to have a not-small set of reverse dependencies, seems actively maintained and skimming the code looks reasonable...